ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/QueryStreamHandler.java

```diff
@@ -281,13 +292,63 @@ private void handleQueryPublisher(
     queryPublisher.subscribe(querySubscriber);
   }
 
+  private void handlePrintPublisher(
+      final RoutingContext routingContext,
+      final BlockingPrintPublisher printPublisher,
+      final MetricsCallbackHolder metricsCallbackHolder,
+      final long startTimeNanos
+  ) {
+    // The end handler can be called twice if the connection is closed by the client.  The
+    // call to response.end() resulting from queryPublisher.close() may result in a second
+    // call to the end handler, which will mess up metrics, so we ensure that this called just
+    // once by keeping track of the calls.
+    final AtomicBoolean endedResponse = new AtomicBoolean(false);
+    // When response is complete, publisher should be closed
+    routingContext.response().endHandler(v ->
+        endhandler(
+            routingContext,
+            printPublisher,
+            endedResponse,
+            metricsCallbackHolder,
+            startTimeNanos
+        ));
+
+    DelimitedPrintResponseWriter printResponseWriter = new DelimitedPrintResponseWriter(
+        routingContext.response()
+    );
+    PrintSubscriber printSubscriber = new PrintSubscriber(
+        context,
+        routingContext.response(),
+        printResponseWriter
+    );
+
+    printPublisher.subscribe(printSubscriber);
+  }
+
+  private void endhandler(RoutingContext routingContext,
+      BlockingPrintPublisher printPublisher, AtomicBoolean endedResponse,
+      MetricsCallbackHolder metricsCallbackHolder, long startTimeNanos) {
+    if (endedResponse.getAndSet(true)) {
+      log.warn("Connection already closed so just returning");
+      return;
+    }
+
+    printPublisher.close();
+    metricsCallbackHolder.reportMetrics(
```

Comment:
No metrics callbacks are ever registered for PRINT TOPIC statements from this endpoint, so let's remove this.